ui/src/androidTest/java/com/hyperwallet/android/transfermethod/ui/ListReceiptsTest.java

```diff
@@ -0,0 +1,175 @@
+package com.hyperwallet.android.transfermethod.ui;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withParent;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+
+import static java.net.HttpURLConnection.HTTP_NO_CONTENT;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import static com.hyperwallet.android.util.EspressoUtils.atPosition;
+
+import android.widget.TextView;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+
+import com.hyperwallet.android.Hyperwallet;
+import com.hyperwallet.android.hyperwallet_ui.R;
+import com.hyperwallet.android.receipt.view.ListReceiptActivity;
+import com.hyperwallet.android.rule.HyperwalletExternalResourceManager;
+import com.hyperwallet.android.rule.HyperwalletMockWebServer;
+import com.hyperwallet.android.ui.repository.RepositoryFactory;
+import com.hyperwallet.android.util.RecyclerViewCountAssertion;
+import com.hyperwallet.android.util.TestAuthenticationProvider;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class ListReceiptsTest {
+
+    @ClassRule
+    public static HyperwalletExternalResourceManager sResourceManager = new HyperwalletExternalResourceManager();
+    @Rule
+    public HyperwalletMockWebServer mMockWebServer = new HyperwalletMockWebServer(8080);
+    @Rule
+    public ActivityTestRule<ListReceiptActivity> mActivityTestRule =
+            new ActivityTestRule<>(ListReceiptActivity.class, true, false);
+
+    @Before
+    public void setup() {
+        Hyperwallet.getInstance(new TestAuthenticationProvider());
+
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("authentication_token_response.json")).mock();
+    }
+
+    @After
+    public void cleanup() {
+        RepositoryFactory.clearInstance();
+    }
+
+    @Test
+    public void testListReceipts_userHasMultipleTransactions() {
```

Comment:
Please add test case for Locale/Timezone sensitive information, since normal Unit test will not change the presentation of strings 
for example
@before -> set Locale.setDefault(Locale.US) 
then in test
set Locale.setDefault(Locale.JAPAN) 
this will change the display of dates

Reply:
@peter-joseph I added Italy